module/sockets.mjs

```diff
@@ -0,0 +1,67 @@
+import sockets from "./sockets/_module.mjs";
+
+/**
+ * @typedef {object} SocketEventConfig
+ * @property {string} event           The unique socket event name.
+ * @property {Function} initiate      A function used to initiate the socket event.
+ * @property {Function} finalize      A function that finalizes the socket event as the valid user(s).
+ */
+
+export default class Sockets5e {
+  constructor() {
+    game.socket.on("system.dnd5e", this.#handleSocket.bind(this));
+    for (const config of sockets) this.#register(config);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Store event configurations.
+   * @type {Map<string, SocketEventConfig>}
+   */
+  #events = new Map();
+
+  /* -------------------------------------------- */
+
+  /**
+   * Register a sovket event config.
+   * @param {SocketEventConfig} config
+   */
+  #register(config) {
+    this.#events.set(config.event, config);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Handle the socket event for the correct user(s).
+   * @param {object} data              Socket data.
+   * @param {boolean} [data._emit]     Internally used parameter for whether this was initiated or emitted.
+   * @returns {*}
+   */
+  #handleSocket({ _emit, ...data }) {
+    if ( !data.userIds.includes(game.user.id) ) {
+      if (!_emit) game.socket.emit("system.dnd5e", { ...data, _emit: true });
+      else return;
```

Comment:
Seems like this shouldn't be in an `else`, otherwise it causes emits the socket event and then also attempts to trigger locally.

Reply:
Literally went through the logic several times.

Gosh darn.